pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -2152,8 +2159,16 @@ static void validateRequest(PinotQuery pinotQuery, int queryResponseLimit) {
       throw new IllegalStateException("SQL query should always have response format and group-by mode set to SQL");
     }
 
+    // throw errors if options is less than 1, rectify if larger that current replicas
+    if (queryOptions.get(Broker.Request.QueryOptionKey.NUM_REPLICA_GROUPS) != null) {
```

Comment:
I'm not sure if the `validateRequest()` is the correct place to update value for `queryOptions`. Why don't we process this in `select()`?

Reply:
Do you mean in setOptions? or in select in InstanceSelector? In that case what should validate do?

Reply:
Also If we do this in select, we will have to store tableConfig in instanceSelector to get the correct number of replicas or pass that as an extra information?

Reply:
no longer updating the values in validateRequest, it will throw error for invalid options